### PR TITLE
55 from ds

### LINKED ISSUE #55: `from_zarr` function
Read a zarr file, get complex, phase, amplitude as lazy attributes.

In the recent dev in PyDePSI, we made functionalities to convert binary inteferograms files to zarr, and the processing can start from zarr files (see [example here](https://github.com/MotionbyLearning/PyDePSI/blob/80ef2e3ac72b7bd88bd664ae7ccabfcef7c8826c/examples/scripts/script_ps_selection.py), in Line 96-101). The only problem is we need three variables: `complex`, `amplitude`, `phase` for processing. In `from_binary` function, they are created as lazy variables. In zarr file, `real` and `imag` fields are created, so like-wise, a function `from_zarr` is needed to create task graph to generate these three variables from Zarr.

### PR DESCRIPTION
**Pull request by rogerkuou**
Fix #55.

- [x] Add `from_zarr` function
- [x] Add test files and unit tests
- [x] Replace lint unit test with precommit hooks
- [x] Remove `example` dir from ruff check

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost, when you have time, can you review this PR? The details should be in the issue description.

While creating this PR, the ruff linting is failling on the notebooks. These notebooks exists mainly for archiving purposes. In the future they will all be moved to [PyDePSI workflow](https://github.com/MotionbyLearning/PyDePSI/tree/main/examples). Therefore I turned off ruff linting on them. I also replaced the unit test for linting with a pre-commit hook.

File: sarxarray/_io.py [hunk 1]
```diff
@@ -13,6 +14,60 @@
 # Example: https://docs.dask.org/en/stable/array-creation.html#memory-mapping
 
 
+def from_zarr(files_zarr: str | Path) -> xr.Dataset:
+    """Read a SLC stack or relabted variables from zarr files.
```
Comment (SarahAlidoost) [inline sarxarray/_io.py:18]:
```suggestion
    """Read a SLC stack or related variables from zarr files.
```

Comment (SarahAlidoost) [reply sarxarray/_io.py:18]: It is not clear what "or related variables" means. This function does not support selecting only one variable to read from a zarr file, right? I suggest removing "or related variables".

File: sarxarray/_io.py [hunk 2]
```diff
@@ -13,6 +14,60 @@
...
+    ------
+    ValueError
+        The input dataset should have three dimensions: (azimuth, range, time).
+    ValueError
+        The input dataset should have the following variables: ('real', 'imag').
+    """
+    ds = xr.open_zarr(files_zarr)
+
+    # Check ds should have the following dimensions: (azimuth, range, time)
+    if not all([dim in ds.dims for dim in ["azimuth", "range", "time"]]):
```
Comment (SarahAlidoost) [inline sarxarray/_io.py:49]:
```suggestion
    if any([dim not in ds.dims for dim in ["azimuth", "range", "time"]]):
```

File: sarxarray/_io.py [hunk 3]
```diff
@@ -13,6 +14,60 @@
...
+        raise ValueError(
+            "The input dataset should have three dimensions: (azimuth, range, time)."
+        )
+
+    # Check ds should have the following variables: ("real", "imag")
+    if not all([var in ds.variables for var in ["real", "imag"]]):
```
Comment (SarahAlidoost) [inline sarxarray/_io.py:55]:
```suggestion
    if any([var not in ds.variables for var in ["real", "imag"]]):
```

File: sarxarray/_io.py [hunk 4]
```diff
@@ -13,6 +14,60 @@
 # Example: https://docs.dask.org/en/stable/array-creation.html#memory-mapping
 
 
+def from_zarr(files_zarr: str | Path) -> xr.Dataset:
```
Comment (SarahAlidoost) [inline sarxarray/_io.py:17]: The `from_zarr` function currently wraps the [`xarray.open_zarr`](https://docs.xarray.dev/en/latest/generated/xarray.open_zarr.html) function but does not expose its additional arguments, which restricts users' flexibility when using `from_zarr`. I suggest adding `**kwargs` to allow users to specify other arguments for `open_zarr`, or at least, expose some commonly used parameters e.g. `chunks`, `drop_variables`, and `zarr_version`, ....

File: tests/test_io.py [hunk 5]
```diff
@@ -10,7 +11,36 @@
...
+
+
+class TestFromZarr:
+    """from_zarr in _io.py"""
+
+    def test_from_zarr_normal(self):
+        slcs = sarxarray.from_zarr(
+            f"{os.path.dirname(__file__)}/data/zarrs/slcs_example.zarr"
+        )
+        assert all([dim in slcs.dims for dim in ["azimuth", "range", "time"]])
```
Comment (SarahAlidoost) [inline tests/test_io.py:27]:
```suggestion
        assert all(dim in slcs.dims for dim in ["azimuth", "range", "time"])
```

File: tests/test_io.py [hunk 6]
```diff
@@ -10,7 +11,36 @@
...
+        assert all([var not in slcs.variables.keys() for var in ["real", "imag"]])
```
Comment (SarahAlidoost) [inline tests/test_io.py:28]:
```suggestion
        assert all(var not in slcs.variables.keys() for var in ["real", "imag"])
```

File: tests/test_io.py [hunk 7]
```diff
@@ -10,7 +11,36 @@
...
+        assert all(
+            [var in slcs.variables.keys() for var in ["complex", "amplitude", "phase"]]
```
Comment (SarahAlidoost) [inline tests/test_io.py:30]:
```suggestion
            [var in slcs.variables.keys() for var in ["complex", "amplitude", "phase"]]
```
```suggestion
            var in slcs.variables.keys() for var in ["complex", "amplitude", "phase"]
```

**Review by SarahAlidoost — CHANGES_REQUESTED** [review]
@rogerkuou please see my comments and suggestions, the main concern is [this comment](https://github.com/TUDelftGeodesy/sarxarray/pull/56/files#r1784319729).

**Comment by rogerkuou** [comment]
Thanks @SarahAlidoost! I adpted your suggestions and re-write `from_zarr` to `from_ds`. Now the reading function operates on a DS read by `open_zarr`. Can you give it another look?

**Review by SarahAlidoost — APPROVED** [review]
@rogerkuou thanks for the fixes, looks great :+1:

**Pull request merged**